### PR TITLE
Fix destruction of articulations

### PR DESCRIPTION
**Pull request by arturkamieniecki**
## What does this PR do?

This PR resolves https://github.com/o3de/o3de/issues/16990, by adding a handler that destroys articulations if the scene is removed. It also prevents a crash during the deactivation of the component by checking if the scene is a `nullptr`.

## How was this PR tested?

This PR was tested by reproducing the behavior described in https://github.com/o3de/o3de/issues/16990.

### CODE REVIEW

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 1]
```diff
@@ -126,6 +126,19 @@ namespace PhysX
             {
                 sceneInterface->RegisterSceneSimulationFinishHandler(m_attachedSceneHandle, m_sceneFinishSimHandler);
 
+                // Create a handler that in the case that the scene was removed before the deactivation of the component,
+                // ensures that all articulations are destroyed.
+                m_sceneRemovedHandler = AzPhysics::SystemEvents::OnSceneRemovedEvent::Handler(
+                    [this]([[maybe_unused]] AzPhysics::SceneHandle sceneHandle)
```
Comment (AMZN-alexpete) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:132]: sceneHandle appears to be used
```suggestion
                    [this](AzPhysics::SceneHandle sceneHandle)
```

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 2]
```diff
@@ -469,6 +484,12 @@ namespace PhysX
     void ArticulationLinkComponent::DestroyArticulation()
     {
         AzPhysics::Scene* scene = AZ::Interface<AzPhysics::SceneInterface>::Get()->GetScene(m_attachedSceneHandle);
+        if (scene == nullptr)
+        {
+            // The scene can be removed before articulation is destroyed.
+            // If the scene was removed. Articulations were also removed.
+            return;
+        }
```
Comment (AMZN-alexpete) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:492]: does `m_articulationLinks` still get cleared when the scene is removed?

Comment (lumberyard-employee-dm) [reply Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:492]: Yeah this check should be scoped to be only around the scene and pxScene statements.

Comment (arturkamieniecki) [reply Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:492]: This check prevents the function from running twice. If the scene was removed this function would run due to the handler and after the component will be deactivated, not running the `DestroyArticulation` function (as the scene at that point would be nullptr). If the scene wasn't removed the handler gets disconnected and again this function runs normally.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by michalpelka — APPROVED** [review]
Great work @arturkamieniecki !

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**